hv-rhel6.x/hv/pci-hyperv.c

```diff
@@ -2467,27 +2475,54 @@ static int hv_pci_remove(struct hv_device *hdev)
 			       VMBUS_DATA_PACKET_FLAG_COMPLETION_REQUESTED);
 	if (!ret)
 		wait_for_completion_timeout(&comp_pkt.host_event, 10 * HZ);
+}
 
+void pci_stop_remove_root_bus(struct pci_bus *bus)
+{
+        struct pci_dev *child, *tmp;
+        struct pci_host_bridge *host_bridge;
+
+        if (!pci_is_root_bus(bus))
+                return;
+
+        host_bridge = to_pci_host_bridge(bus->bridge);
+        list_for_each_entry_safe_reverse(child, tmp,
+                                         &bus->devices, bus_list)
+                pci_stop_and_remove_bus_device(child);
+
+       /* stop the host bridge */
+        device_del(&host_bridge->dev);
+
+        pci_remove_bus(bus);
+        host_bridge->bus = NULL;
+
+        /* remove the host bridge */
+        put_device(&host_bridge->dev);
+}
+
+/**
+ * hv_pci_remove() - Remove routine for this VMBus channel
+ * @hdev:	VMBus's tracking struct for this root PCI bus
+ *
+ * Return: 0 on success, -errno on failure
+ */
+static int hv_pci_remove(struct hv_device *hdev)
+{
+	struct hv_pcibus_device *hbus;
+
+	hbus = hv_get_drvdata(hdev);
 	if (hbus->state == hv_pcibus_installed) {
 		/* Remove the bus from PCI's point of view. */
 		pci_lock_rescan_remove();
-//		pci_stop_root_bus(hbus->pci_bus);
-//		pci_remove_root_bus(hbus->pci_bus);
+		pci_stop_remove_root_bus(hbus->pci_bus);
```

Comment:
Use 2 API's pci_stop_root_bus() & pci_remove_root_bus() to make it consistent with RH7.X